File: EMarket/src/main/java/DoAn/B19DCCN445/EMarket/model/Order_detail.java
package DoAn.B19DCCN445.EMarket.model;


import java.util.Date;

import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
@Data
@Entity
@Builder
@EntityListeners(AuditingEntityListener.class)
@NoArgsConstructor
@AllArgsConstructor
//@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Order_detail {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private int quantity;
	private Double unit_price;
	private Double discount_percentage;
	private Double discount_amount;
	private String order_detail_status;
	private Date date_allocated;
	
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "order_id")
	@JsonBackReference("order-Order_detail")
	private Order order;
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "product_id")
	@JsonBackReference("product-Order_detail")
	private Product product;
}

File: emarket-ui/src/models/order_detail.js
import Order from "./order";
import Product from "./product";

export default class Order_detail {
    constructor(
        id = null,
        quantity = 0,
        unit_price,
        discount_percentage,
        discount_amount,
        order_detail_status,
        date_allocated,
        order = new Order(),
        product = new Product(),
        isReviewed
    ) {
        Object.assign(this, {
            id,
            quantity,
            unit_price,
            discount_percentage,
            discount_amount,
            order_detail_status,
            date_allocated,
            order,
            product,
            isReviewed
        });
    }
}
